docs/understanding-your-pipeline/canonical-event/index.md

```diff
@@ -42,280 +48,411 @@ The sections below go over the standard fields found in all Snowplow events.
+
+```sql
+select
+...
+unstruct_event_my_example_event_1_0_0:myCustomField::varchar, -- field will be variant type so important to cast
+...
+from 
+    atomic.events
+```
+
+</TabItem>
+<TabItem value="db" label="Databricks">
 
-| **Field** | **Type** | **Description** | **Reqd?** | **Example** |
-| --- | --- | --- | --- | --- |
-| `se_category` | text | Category of event | Yes | 'ecomm', 'video' |
-| `se_action` | text | Action performed / event name | Yes | 'add-to-basket', 'play-video' |
-| `se_label` | text | The object of the action e.g. the ID of the video played or SKU of the product added-to-basket | No | 'pbz00123' |
-| `se_property` | text | A property associated with the object of the action | No | 'HD', 'large' |
-| `se_value` | decimal | A value associated with the event / action e.g. the value of goods added-to-basket | No | 9.99 |
+The unstructured event is not part of the standard `atomic.events` columns, it is instead a `struct` type column with the name of the schema and the major version number. The fields in this table will be determined by the JSON schema defined for the event in advance, converted to snake case. Users can query fields in the self-describing event by extracting them using 
 
-#### Custom unstructured events
+```sql
+select
+...
+unstruct_event_my_example_event_1_0_0.myCustomField,
+...
+from 
+    atomic.events
+```
+
+</TabItem>
+</Tabs>
+
+
+#### Entities (Contexts)
+
+Entities (Contexts) enable Snowplow users to define their own entities that are related to events, and fields that are related to each of those entities. For example, an online retailer may choose to define a `user` context, to store information about a particular user, which might include data points like the users Facebook ID, age, membership details etc. In addition, they may also define a `product` context, with product data e.g. SKU, name, created date, description, tags etc.
+
+An event can have any number of custom entities attached. Each context is passed into Snowplow as a JSON. Additionally, the [Snowplow Enrichment](/docs/enriching-your-data/index.md) process can derive additional contexts.
 
-:::note
+<Tabs groupId="warehouse" queryString>
+<TabItem value="rs/pg" label="Redshift/Postgres" default>
 
-Currently, custom events are not available in [BDP Cloud](/docs/getting-started-with-snowplow-bdp/cloud/index.md).
+Contexts are not part of the `atomic.events` table; instead, for users running on Redshift, Snowplow will shred each context JSON into a dedicated table in the `atomic` schema, making it much more efficient for analysts to query data passed in in any one of the contexts. Those contexts can be joined back to the core `atomic.events` table by the following, which is a one-to-one join or a one-to-many join.
```

Comment:
I'd spell this out here (although it is kind of implicit just not the order) that this means one row in `events` can be 0 or more rows in a context table.